src/negotiation/maker_client.ts

```diff
@@ -0,0 +1,31 @@
+import axios from "axios";
+import { ExecutionParams } from "./execution_params";
+import { OrderParams } from "./order";
+
+export class MakerClient {
+  constructor(private readonly makerUrl: string) {}
+
+  public async getOrderByTradingPair(
+    tradingPair: string
+  ): Promise<OrderParams> {
+    const response = await axios.get(`${this.makerUrl}orders/${tradingPair}`);
+    return response.data;
+  }
+
+  public async getExecutionParams(
+    order: OrderParams
+  ): Promise<ExecutionParams> {
+    const response = await axios.get(
+      `${this.makerUrl}orders/${order.id}/executionParams`
+    );
+    return response.data;
+  }
+
+  public async takeOrder(order: OrderParams, swapId: string) {
```

Comment:
❓ How come we already have a `swapId` at this point? Aren't we supposed to get the `swapId` from the maker **after** taking the order?

Reply:
This is the swap Id indeed. I have documented in f622aea0